build.gradle

```diff
@@ -28,70 +28,324 @@
-    release = isSnapshot ? "0.1" : '1'
-    version = "${project.version}"
+tasks.withType(JavaCompile) {
+    options.encoding = 'UTF-8'
+}
+
+static def getTimestamp() {
+    def df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
+    df.setTimeZone(TimeZone.getTimeZone("UTC"))
+    return df.format(new Date())
+}
 
-    into '/usr/share/opensearch/plugins'
-    from(zipTree(project.file(archivePath).absolutePath)) {
-        into "opensearch-security"
-        permissionGroup 'opensearch'
+static def gitCommitId() {
+    def cmd = "git rev-parse HEAD"
+    def proc = cmd.execute()
+    return proc.text.trim()
+}
+
+jar {
+    manifest {
+        attributes(
+                "Manifest-Version": "1.0",
+                "Created-By": "Gradle ${gradle.gradleVersion}",
+                "Build-Jdk": "${System.properties['java.version']}",
+                "Implementation-Title": "OpenSearch Security",
+                "Implementation-Version": archiveVersion,
+                "Implementation-Vendor-Id": "org.opensearch",
+                "Implementation-URL": "https://github.com/opensearch-project/security",
+                "Build-Time": getTimestamp(),
+                "Built-By": "OpenSearch Security Plugin",
+                "git-sha1": gitCommitId()
+        )
     }
 
-    user 'root'
-    permissionGroup 'root'
+    libsDirName = '.'
+    into '', {
+        from 'NOTICE.txt', "THIRD-PARTY.txt", "LICENSE"
+    }
+    processResources {
+        exclude("KEYS")
+    }
 
-    requires('opensearch-oss', "1.3.0", EQUAL)
-    packager = 'Amazon'
-    vendor = 'Amazon'
-    os = 'LINUX'
-    prefix '/usr'
+}
+
+testsJar {
+    manifest {
+        attributes(
+                "Manifest-Version": "1.0",
+                "Created-By": "Gradle ${gradle.gradleVersion}",
+                "Build-Jdk": "${System.properties['java.version']}",
+                "Implementation-Title": "OpenSearch Security",
+                "Implementation-Version": archiveVersion,
+                "Implementation-Vendor-Id": "org.opensearch",
+                "Implementation-URL": "https://github.com/opensearch-project/security",
+                "Build-Time": getTimestamp(),
+                "Built-By": "OpenSearch Security Plugin",
+                "git-sha1": gitCommitId()
+        )
+    }
+
+    libsDirName = '.'
+}
+
+boolean isCI = System.getenv().containsKey("CI")
 
-    license 'ASL-2.0'
-    maintainer 'OpenDistro for Elasticsearch Team <opendistro@amazon.com>'
-    url 'https://opendistro.github.io/for-elasticsearch/downloads.html'
-    summary '''
-     OpenSearch Security plugin. 
-     Reference documentation can be found at https://opendistro.github.io/for-elasticsearch-docs/.
-     '''.stripIndent().replace('\n', ' ').trim()
+test {
+    maxParallelForks = 3
+    jvmArgs "-Xmx3072m"
+    retry {
+        if (isCI) {
+            failOnPassedAfterRetry = false
+            maxFailures = 30
+            maxRetries = 3
+        }
```

Comment:
Any reason why this would be different in CI? When I write code I expect my test runs to match what happens in CI and not pass less/more frequently.

Reply:
I wanted to be fully compliant with old maven build and I wrongly assumed that it (retries) was enabled only for CI.

I've just removed `-DCI=true` part